### ISSUE TITLE
Merlin 4.11 sometimes finds a module instead of a constructor

### ISSUE THREAD

**Issue by antalsz** [ocaml/merlin#1278]
The following file has a module and a constructor both named `C`:

```ocaml
module C = struct end

module M = struct
  type t = C
end

let (_ : M.t) = C
```

On the last line, OCaml disambiguates `C` to `M.C` thanks to the type of the context.  However, with the cursor at the end of the line, Merlin reports `sig end`, the type of `module C`, instead of `M.t`.

Example command, assuming that the code is in the file `test.ml`:
```shell
ocamlmerlin single type-enclosing -protocol jquery -verbosity 0 -filename test.ml -position 7:17 -index 0 < test.ml | jq '.value[0]'
```
Result:
```json
{
  "start": {
    "line":7,
    "col": 16
  },
  "end": {
    "line": 7,
    "col": 17
  },
  "type": "sig end",
  "tail": "no"
}
```

Furthermore, Merlin will instead correctly report `M.t` if either of the following changes are made:

* If the cursor is positioned elsewhere in the constructor name, in this case at position `7:16`, we get `M.t`.
* If the last line is duplicated, then leaving the cursor on this line (`7:17`) produces `M.t`; in this case, only the end of the file (`8:17`) incorrectly reports `sig end`.  (On both lines, moving earlier in the constructor also fixes the issue; both `7:16` and `8:16` also produce `M.t`.)

**Comment by voodoos** [comment]
It looks like this issue was fixed with our last reelase (4.2 / 3.5.0).
I added your test to the suite: https://github.com/ocaml/merlin/commit/1ad950191461c1d0b560b97b1c8f9223eadde57f

Can you confirm that ?

**Comment by antalsz** [comment]
Seems good!  Thank you!
